app/models/legal_aid_application.rb

```diff
@@ -116,4 +117,31 @@ def create_app_ref
   def set_open_banking_consent_choice_at
     self.open_banking_consent_choice_at = Time.current if will_save_change_to_open_banking_consent?
   end
+
+  def keep_attributes_in_sync
+    reset_to_match_own_home_no
+    reset_to_match_own_home_mortgage
+    reset_to_match_shared_ownership_no
+  end
+
+  def reset_to_match_own_home_no
+    return unless own_home_changed? && own_home_no?
+
+    self.property_value = nil
+    self.outstanding_mortgage_amount = nil
+    self.shared_ownership = nil
+    self.percentage_home = nil
+  end
+
+  def reset_to_match_own_home_mortgage
+    return unless own_home_changed? && own_home_mortgage?
+
+    self.outstanding_mortgage_amount = nil
```

Comment:
should the condition be `!own_home_mortgage?` instead of `own_home_mortgage` ?

Reply:
I think refactor has fixed the logic.